internal/vault/client_test.go

```diff
@@ -4,14 +4,46 @@
 package vault
 
 import (
+	"context"
+	"crypto/x509"
 	"fmt"
 	"testing"
 	"time"
 
 	"github.com/hashicorp/vault/api"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	secretsv1alpha1 "github.com/hashicorp/vault-secrets-operator/api/v1alpha1"
+	"github.com/hashicorp/vault-secrets-operator/internal/consts"
+	"github.com/hashicorp/vault-secrets-operator/internal/vault/credentials"
 )
 
+const testCACertPEM = `-----BEGIN CERTIFICATE-----
```

Comment:
Would [generateCA()](https://github.com/hashicorp/vault-secrets-operator/blob/c9f007e732bfad9b8003f00f518aad253287a2fa/internal/vault/tls_test.go#L25) work for this instead of hard-coding the test certificate?